src/main/react/app/stores/flow-store.ts

```diff
@@ -60,6 +72,22 @@ const useFlowStore = create<FlowState>((set, get) => ({
   setEdges: (edges) => {
     set({ edges })
   },
+  getNextNodeId: () => {
+    const current = get().nodeIdCounter
+    set({ nodeIdCounter: current + 1 })
+    return current.toString()
+  },
+  addNode: (newNode: FlowNode) => {
+    set({
+      nodes: [...get().nodes, newNode],
+    })
+  },
+  deleteNode: (nodeId: string) => {
+    set({
+      nodes: get().nodes.filter((node) => node.id !== nodeId),
+      edges: get().edges.filter((edge) => edge.source !== nodeId && edge.target !== nodeId),
```

Comment:
These things need to be an array for the ReactFlow library?

Reply:
Yes I saw. I was just wondering, because otherwise stuff could be deleted by doing something like: 
```ts
delete node[id]
``` 
Without having to iterate over the list. But ok if it is a library limitation.